core/src/main/java/com/linecorp/armeria/internal/client/PublicSuffix.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.client;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+/**
+ * Utility class to determine if a domain is a public suffix. List of rules taken from
+ * <a href="https://publicsuffix.org/list/public_suffix_list.dat">Public Suffix List</a>.
+ */
+public final class PublicSuffix {
```

Comment:
Guava has [InternetDomainName](https://github.com/google/guava/wiki/InternetDomainNameExplained) that also provides a public suffix checker.
If you don't mind, could you measure performance between this and Guava's?